### PR TITLE
Coding: Develop a timeline for programs page (#73)

### LINKED ISSUE #73: Coding: Develop a timeline of recent programs for the Programs Page
As a developer,
I need a recent programs timeline on Programs Page to be developed.
The timeline should follow the [Web Metrics Guide.](https://docs.google.com/document/d/1HhcXfc1qSj8aoj31psUgfdecI6zTkkzxUvueiEu4eas/edit) 

[Programs Page Mock](https://drive.google.com/drive/folders/1QSFH6e1WqJ1t--O_cQa-VS6o6zpF3xeG)
 
**Timeline -**

programs-timeline

**Acceptance criteria :**
- [ ] PR must follow the Web Metrics Guide

**Definition of Done**
- [ ] All of the required items are completed.
- [ ]  Approval by 1 mentor.

Estimation
1-3 days

### PR DESCRIPTION
**Pull request by iamkryptonite**
### Description

Developed a timeline for the events page. 

Fixes #73

### Type of Change:
**Delete irrelevant options.**

- Code
- User Interface

**Code/Quality Assurance Only**
- Bugfix (a non-breaking change that fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.
AnitaB_org_open_source

ezgif com-video-to-gif

**gh-pages Link:** https://iamkryptonite.github.io/anitab-org.github.io/

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented on my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: src/Components/Programs/Timeline.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import React, {useEffect} from 'react';
+import {ScrollView,Text,View,TouchableWithoutFeedback} from 'react-native';
+import {Event,Line,Date,Marker,Fade,Stroke} from './styles';
+import $ from 'jquery';
+import {ArrowLeftOutlined ,ArrowRightOutlined} from '@ant-design/icons'
+import dayjs from 'dayjs'
+const events=[
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:7]: this should be kept in content js so we can edit

File: src/Components/Programs/Timeline.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        scroll.scrollLeft(p-=35);
+        if(p<0)
+        p=0;
+    }
+    const scrollRight=()=>{
+        var scroll = $('.css-view-1dbjc4n');
+        scroll.scrollLeft(p+=35);
+    }
+  return (
+      <>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:37]: what this line do?

File: src/Components/Programs/Timeline.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                                    <Marker style={{left:(p-1)*5+2+'%'}}></Marker>
+                                </View>
+                            ))
+                        }
+                        <View style={{left:(p-1)*5+3+'%',marginBottom:'15px'}}><Text>Today {dayjs().date()}th October</Text></View>
+                        <View style={{flexDirection:'row'}}>
+                            {
+                                months.map((m)=>(
+                                    <>
+                                        <View style={{left:(m[0]*30)*5+1+'%',marginBottom:'15px',top:'-44px'}}><Text>1st {m[1]}</Text></View>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:88]: styles should be hold using StyleSheet imported from react native

File: src/Components/Programs/Timeline.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                </View>
+                <ScrollView
+                horizontal={true}
+                style={{ marginBottom:'5vw',marginTop:'3vw',flexDirection:'column',width:'70%',position:'absolute',right:'0',top:'-45px'}}
+                showsHorizontalScrollIndicator={false}
+                >
+                    <View style={{width:'90%'}}>
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'3vw',flexDirection:'column'}} key={item.event}>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:71]: 3vw will break native app ones we move it there.

**Comment by annabauza** [comment]
this looks great we just need to tweak the code a bit

File: src/Components/Programs/Timeline.js (hunk 2 above)
Comment (iamkryptonite) [reply src/Components/Programs/Timeline.js:37]: It handles the onClick events . When the user clicks the past or future arrows, it scrolls the timeline accordingly. That long className targets the scrollview.

File: src/Components/Programs/Timeline.js (hunk 1 above)
Comment (iamkryptonite) [reply src/Components/Programs/Timeline.js:7]: okay

File: src/Components/Programs/Timeline.js (hunk 3 above)
Comment (iamkryptonite) [reply src/Components/Programs/Timeline.js]: I have made some changes although there are still some inline styles which cannot be changed .

**Comment by iamkryptonite** [comment]
Hey @annabauza I have made the necessary changes in the code ,could you please review it once

**Comment by nandini45** [comment]
@Bucky25 can you review this one?

File: src/Components/Programs/Timeline.js [hunk 5]
```diff
@@ -62,13 +55,13 @@ function Timeline() {
...
                 horizontal={true}
-                style={{ marginBottom:'5vw',marginTop:'3vw',flexDirection:'column',width:'70%',position:'absolute',right:'0',top:'-45px'}}
+                style={{ marginBottom:'5vw',marginTop:'45px',flexDirection:'column',width:'70%',position:'absolute',right:'0',top:'-45px'}}
                 showsHorizontalScrollIndicator={false}
                 >
                     <View style={{width:'90%'}}>
                         {
                             events.map((item)=>(
-                                <View style={{marginBottom:'3vw',flexDirection:'column'}} key={item.event}>
+                                <View style={{marginBottom:'47px',flexDirection:'column'}} key={item.event}>
```
Comment (Bucky25) [inline src/Components/Programs/Timeline.js:64]: please make it  a power of 2

File: src/Components/Programs/styles.js [hunk 6]
```diff
@@ -58,6 +57,35 @@ export const Stroke = styled(View,{
...
+    width:'70%',
+    position:'absolute',
+    right:'0',
+    top:'-3px'
+})
+export const Container = styled(View,{
+    width:'80%'
+})
+export const Months = styled(View,{
+    marginBottom:'15px',
```
Comment (Bucky25) [inline src/Components/Programs/styles.js:89]: please make it power of 2

File: src/Components/Programs/styles.js [hunk 7]
```diff
@@ -58,6 +57,35 @@ export const Stroke = styled(View,{
...
     top:'-35px',
-    // zIndex:'10px'
-
-})
\ No newline at end of file
+})
+export const ArrowNavigation = styled(View,{
+    flexDirection:'row',
+    marginTop:'10px',
+    marginBottom:'10px',
```
Comment (Bucky25) [inline src/Components/Programs/styles.js:64]: please make it power of 2

**Review by Bucky25 — CHANGES_REQUESTED** [review]

File: src/Components/Programs/styles.js (hunk 7 above)
Comment (Bucky25) [reply src/Components/Programs/styles.js:64]: and please add a link to your deployed gh pages

File: src/Components/Programs/styles.js (hunk 6 above)
Comment (iamkryptonite) [reply src/Components/Programs/styles.js:89]: Okay

File: src/Components/Programs/styles.js (hunk 7 above)
Comment (iamkryptonite) [reply src/Components/Programs/styles.js]: gh-pages link: https://iamkryptonite.github.io/anitab-org.github.io/

File: src/Components/Programs/styles.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
+import { View, Text ,ScrollView} from 'react-native';
+import { styled } from 'react-native-reflect';
+export const Event = styled(Text,{
+    fontSize: 24,
+    borderWidth: 0.5,
+    paddingVertical:4,
+    paddingHorizontal: 8,
+    marginBottom: 4,
+    borderRadius: '50px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:9]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    textAlign: 'center',
+    zIndex:'1px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:11]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    position:'relative',
+    backgroundColor:'#fff',
+    width:'200px'
```
Comment (annabauza) [inline src/Components/Programs/styles.js:14]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+});
+export const Line = styled(View,{
+    width:'200vw',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:17]: we don't use vw because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    borderBottomWidth:10,
+    marginBottom:4,
+    position:'relative',
+    top: '18px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:21]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    left: '20%',
+});
+export const Date = styled(View,{
+    backgroundColor:'#ffdd30',
+    color:'#fff',
+    fontSize:20,
+    width:45,
+    height:45,
+    borderRadius: '50px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:30]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    textAlign: 'center',
+    justifyContent:'center',
+    position:'absolute',
+});
+export const Marker = styled(View,{
+    backgroundColor:'#000',
+    width:'2px',
+    height: '60px',
+    position:'absolute',
+    top: '28px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:40]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    width:45,
+    height:45,
+    borderRadius: '50px',
+    textAlign: 'center',
+    justifyContent:'center',
+    position:'absolute',
+});
+export const Marker = styled(View,{
+    backgroundColor:'#000',
+    width:'2px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:37]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    height: '60px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:38]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    position:'absolute',
+    top: '28px',
+    zIndex:'-1px'
+});
+export const Fade = styled(View,{
+    backgroundColor:'#fff',
+    height:'90px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:45]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    position:'absolute',
+    opacity:'0.6',
+})
+export const Stroke = styled(View,{
+    borderBottomColor:'#000',
+    borderBottomWidth:'2px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:51]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    width:'50vw',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:52]: we don't use vw because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+    top:'-45px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:53]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+})
+export const ArrowNavigation = styled(View,{
+    flexDirection:'row',
+    marginTop:'8px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:57]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+    marginBottom:'8px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:58]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+    width:'80%',
+    justifyContent:'space-between'
+})
+export const ArrowText = styled(Text,{
+    fontSize:'16px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:63]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 24]
```diff
@@ -0,0 +1,85 @@
...
+    marginLeft:'8px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:64]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 25]
```diff
@@ -0,0 +1,85 @@
...
+    marginRight:'8px'
```
Comment (annabauza) [inline src/Components/Programs/styles.js:65]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 26]
```diff
@@ -0,0 +1,85 @@
...
+})
+export const ScrollContainer = styled(View,{
+    width:'80%',
+    flexWrap:'wrap',
+    flexDirection:'column',
+    marginBottom:'64px'
```
Comment (annabauza) [inline src/Components/Programs/styles.js:71]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 27]
```diff
@@ -0,0 +1,85 @@
...
+    flexDirection:'column',
+    width:'70%',
+    position:'absolute',
+    right:'0',
+})
+export const Container = styled(View,{
+    width:'80%'
+})
+export const Months = styled(View,{
+    marginBottom:'16px',
```
Comment (annabauza) [inline src/Components/Programs/styles.js:83]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/styles.js [hunk 28]
```diff
@@ -0,0 +1,85 @@
...
+    top:'-35px'
```
Comment (annabauza) [inline src/Components/Programs/styles.js:84]: we don't use px because this will break moving to modules to native app.

File: src/Components/Programs/Timeline.js [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+                    ))
+                }
+                <ScrollMain
+                horizontal={true}
+                showsHorizontalScrollIndicator={false}
+                >
+                    <Container>
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'32px',flexDirection:'column'}} key={item.event}>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:64]: this style can be moved to styles, right?

File: src/Components/Programs/Timeline.js [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+                    <TouchableWithoutFeedback onPress={scrollRight}>
+                        <ArrowRightOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
+                    </TouchableWithoutFeedback>
+                </View>
+            </ArrowNavigation>
+            <ScrollContainer>
+                {
+                    events.map((item)=>(
+                        <View key={item.event}>
+                            <Event style={{color:item.color,borderColor:item.color,flex:1,left:item.date[0][1]+'vw'}}>{item.event}</Event>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:52]: we don't use px or vw, because this will break moving to modules to native app.

File: src/Components/Programs/Timeline.js [hunk 31]
```diff
@@ -0,0 +1,92 @@
...
+                <View style={{flexDirection:'row'}}>
+                    <TouchableWithoutFeedback onPress={scrollLeft}>
+                        <ArrowLeftOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
+                    </TouchableWithoutFeedback>
+                    <ArrowText>PAST</ArrowText>
+                </View>
+                <View style={{flexDirection:'row'}}>
+                    <ArrowText>FUTURE</ArrowText>
+                    <TouchableWithoutFeedback onPress={scrollRight}>
+                        <ArrowRightOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:44]: we don't use px or vw, because this will break moving to modules to native app.

File: src/Components/Programs/Timeline.js [hunk 32]
```diff
@@ -0,0 +1,92 @@
...
+    const scrollRight=()=>{
+        var scroll = $('.css-view-1dbjc4n');
+        scroll.scrollLeft(p+=35);
+    }
+  return (
+      <>
+            <ArrowNavigation>
+                <View style={{flexDirection:'row'}}>
+                    <TouchableWithoutFeedback onPress={scrollLeft}>
+                        <ArrowLeftOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:37]: we don't use px or vw, because this will break moving to modules to native app.

File: src/Components/Programs/Timeline.js [hunk 33]
```diff
@@ -0,0 +1,92 @@
...
+                }
+                <ScrollMain
+                horizontal={true}
+                showsHorizontalScrollIndicator={false}
+                >
+                    <Container>
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'32px',flexDirection:'column'}} key={item.event}>
+                                    <Line style={{left:(item.date[0][1]-1)*5+'%', width:30*12*5+'vw', borderBottomColor:item.color}}></Line>
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:65]: we don't use px or vw, because this will break moving to modules to native app.

File: src/Components/Programs/Timeline.js [hunk 34]
```diff
@@ -0,0 +1,92 @@
...
+import {getEvents,getMonths} from '../../content/programs_events'
+//--------------------------------------------------------------------------------------------
+const events  = getEvents();
+const months = getMonths();
+function Timeline() {
+    var p = dayjs().date(); //current date
+    var curr_month = dayjs().month(); //current month
+    //scrolls the timeline to the current date and month
+    useEffect(() => {
+        var scroll = $('.css-view-1dbjc4n');
```
Comment (annabauza) [inline src/Components/Programs/Timeline.js:16]: is `css-view-1dbjc4n` obfuscated reference?

**Review by annabauza — CHANGES_REQUESTED** [review]
Generally very good code. I have left you mostly notes related to css. Thanks.

**Comment by annabauza** [comment]
@iamkryptonite can you please also resolve your conflicts? Thanks

File: src/Components/Programs/styles.js (hunk 13 above)
Comment (iamkryptonite) [reply src/Components/Programs/styles.js:30]: Without the 'px' border-radius doesn't seem to work. The same problem occurs for z-index also.

Comment (iamkryptonite) [reply src/Components/Programs/styles.js:29]: @annabauza

**Review by tichnas — CHANGES_REQUESTED** [review]
Hi, it looks like the PR will work only for the web version. We should not use things like classes and jQuery in React Native. Please do these changes.
Thanks

**Comment by iamkryptonite** [comment]
> 
> 
> Hi, it looks like the PR will work only for the web version. We should not use things like classes and jQuery in React Native. Please do these changes.
> Thanks

Hi,I have made the changes. Please review once.

**Comment by tichnas** [comment]
@iamkryptonite I think you're still using jQuery and pixels which don't go well with React Native. Can you please change them also?
Thanks for giving time to do these changes :)

**Comment by iamkryptonite** [comment]
> 
> 
> @iamkryptonite I think you're still using jQuery and pixels which don't go well with React Native. Can you please change them also?
> Thanks for giving time to do these changes :)

 Hi @tichnas, I have removed jquery and used useRef instead. And I have also changed all the places where pixels were used.

**Review by annabauza — APPROVED** [review]
💯 This is great thank you!

**Comment by annabauza** [comment]
@iamkryptonite please call yarn in command line and commit yarn.lock because travis is failing.

**Review by tichnas — CHANGES_REQUESTED** [review]
Please commit the changes to `yarn.lock` (and don't delete the file)
Also, remove your homepage link from `package.json` and can you please tell what's `@babel/plugin-transform-react-jsx` used for?

Thanks

**Comment by iamkryptonite** [comment]
> 
> 
> Please commit the changes to `yarn.lock` (and don't delete the file)
> Also, remove your homepage link from `package.json` and can you please tell what's `@babel/plugin-transform-react-jsx` used for?
> 
> Thanks

Hey, I have made the requested changes and also removed the babel package as it was not required.

**Review by annabauza — CHANGES_REQUESTED** [review]
please resolve conflict and merge with existing events tab. Thanks!

File: package.json [hunk 35]
```diff
@@ -1,11 +1,13 @@
 {
-  "homepage": "./",
+  "homepage": "/",
```
Comment (annabauza) [inline package.json:2]: Why this change?

**Review by annabauza — APPROVED** [review]
Generally OK for first iteration just a quick question does homepage change would not break github static pages?

File: package.json (hunk 35 above)
Comment (iamkryptonite) [reply package.json]: I am not really sure, I think this was a mistake so have changed it.

**Review by Vishal-raj-1 — APPROVED** [review]
Great Work @iamkryptonite !! Could you please squash your commits

**Review by annabauza — APPROVED** [review]

**Comment by annabauza** [comment]
@Vishal-raj-1 we are squashing comments on merge, no need to do that.

**Comment by iamkryptonite** [comment]
Hey @annabauza there is one small change that I need to make in my code.

**Comment by annabauza** [comment]
Rise the issue and ask to be assigned

**Comment by iamkryptonite** [comment]
> Rise the issue and ask to be assigned

Hey @annabauza I have raised the [issue](https://github.com/anitab-org/anitab-org.github.io/issues/246). Could you pls approve it and add relevant labels.
Thanks.
